applications/P2PStateChannels.md

```diff
@@ -0,0 +1,254 @@
+	
+Clock
+Every peer has a local subjective logical clock. The clock is synced with the DLT 
+clock (timestamp) in order to provide a local perception of time that mimics the DLT
+time. Peers will never reach consensus on their clocks nor amongst themselves nor     
+with the DLT itself. The protocol is designed to tolerate skewness to a degree. Peers use their clock to decide subjectively, how they want to interact with the system and other peers -  are incoming transactions too far in the future, too far in the past or about in time and based on their subjective opinion they come to an agreement or abstain from signing p2p. They use this clock also to invoke Timeout. While the perception of time is subjective and is designed to tolerate skewness to a degree, once measured and recorded as a timestamp, that information is objective and comparable. While each peer will compare it to their local clock and try to form agreements, if such agreements fail p2p and the system falls back on-chain, the DLT clock will prevail as a single uncontested source of truth and its comparison of the tx timestamp to its clock will be final.
+
+3) AgreementManager (Agreements)
+The system does NOT implement a variation of a classical BFT (Byzantine Fault Tolerant) consensus algorithm due to lack of large economic security amongst peers in a partition and the strong design principle to allow everyone to connect freely with everyone. As such the system can’t guarantee that within a partition there will not be less than ⅓ byzantine actors as required by all classical BFT consensus algorithms. Since the DLT is used as a top level “supreme court”, we can design an optimistic consensus algorithm that tolerates N-1 byzantine actors - in other words we can design a system that’s Byzantine Fault Tolerant with at least 1 honest (non byzantine) actor by ultimately inheriting security from a DLT.
+
+Before explaining optimistic consensus, it’s important to understand the structure of a Transaction in this system:
+
+A transaction in this system references the exact state (previousStateHash) to which it is an input, and references the expected state post execution (stateHash). This resembles the logic behind UTXOs to a degree.
```

Comment:
Are the required to be linearly ordered?  That's slightly more than what UTXOs do.  It's also what you expect when doing full state.  Assuming yes, anything else should proibably be future work.

Reply:
We'll punt on this for now, because the simplest thing alraedy makes sense for initial applications.

Reply:
yes